### PR TITLE
Dependency refresh

### PR DESCRIPTION
**Pull request by weierophinney**
This patch refreshes dependencies as suggested by RenovateBot; they're done in a single PR as most of them needed to be lumped together due to interdependencies, or a need to update the minimum supported PHP version.

Changes include:

- Bumping PHP to 8.1
- Bumping laminas-component-installer to v3 (which included bumping mezzio-tooling to v2.6)
- Bumping Symfony components to v6
- Switching to laminas-coding-standard (from vanilla phpcs)
- Switching to laminas-cli (from home-grown `bin/console` tool)
- Updating to league/commonmark v2.3 (which required minor refactoring of how we consume the component)
- Adding the CI workflow

I've tested all functionality locally, and it worked correctly; I'll be doing a separate PR later with initial tests.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+on:
+  pull_request:
+  push:
+    branches:
+    tags:
+
+jobs:
+  ci:
+    uses: laminas/workflow-continuous-integration/.github/workflows/continuous-integration.yml@1.x
```
Comment (Ocramius) [inline .github/workflows/continuous-integration.yml:11]: I wonder if we can see a CI run on your fork?

Comment (weierophinney) [reply .github/workflows/continuous-integration.yml:11]: Typically, GH has picked these up in new PRs. I'm a bit suprised it's not showing it here. I'll see if it ran on my fork.

Comment (weierophinney) [reply .github/workflows/continuous-integration.yml:11]: Indeed, it did:

- https://github.com/weierophinney/getlaminas.org/actions/runs/2834213442

Comment (Ocramius) [reply .github/workflows/continuous-integration.yml:11]: Excellent! :ship: then :D

**Review by Ocramius — APPROVED** [review]

**Pull request merged**